src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java

```diff
@@ -0,0 +1,701 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import de.terrestris.shogun2.importer.communication.AbstractRESTEntity;
+import de.terrestris.shogun2.importer.communication.RESTData;
+import de.terrestris.shogun2.importer.communication.RESTImport;
+import de.terrestris.shogun2.importer.communication.RESTImportTask;
+import de.terrestris.shogun2.importer.communication.RESTImportTaskList;
+import de.terrestris.shogun2.importer.communication.RESTLayer;
+import de.terrestris.shogun2.importer.communication.RESTTargetDataStore;
+import de.terrestris.shogun2.importer.communication.RESTTargetWorkspace;
+import de.terrestris.shogun2.importer.transform.RESTGdalAddoTransform;
+import de.terrestris.shogun2.importer.transform.RESTGdalTranslateTransform;
+import de.terrestris.shogun2.importer.transform.RESTGdalWarpTransform;
+import de.terrestris.shogun2.importer.transform.RESTReprojectTransform;
+import de.terrestris.shogun2.importer.transform.RESTTransform;
+import de.terrestris.shogun2.util.http.HttpUtil;
+import de.terrestris.shogun2.util.model.Response;
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.exception.ZipException;
+import net.lingala.zip4j.model.FileHeader;
+import net.lingala.zip4j.model.ZipParameters;
+import net.lingala.zip4j.util.Zip4jConstants;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+@Component
+public class GeoServerRESTImporter {
+
+	/**
+	 * The Logger.
+	 */
+	private final static Logger LOG = Logger.getLogger(GeoServerRESTImporter.class);
+
+	/**
+	 *
+	 */
+	private String username;
+
+	/**
+	 *
+	 */
+	private String password;
+
+	/**
+	 *
+	 */
+	private ObjectMapper mapper = new ObjectMapper();
+
+	/**
+	 *
+	 */
+	private URI baseUri;
+
+	/**
+	 *
+	 */
+	public GeoServerRESTImporter() {
+
+	}
+
+	/***
+	 *
+	 * @param importerBaseURL
+	 * @param defaultSRS
+	 * @param username
+	 * @param password
+	 * @throws URISyntaxException
+	 */
+	public GeoServerRESTImporter(String importerBaseURL, String username,
+			String password) throws URISyntaxException {
+		if (StringUtils.isEmpty(importerBaseURL) || StringUtils.isEmpty(username) ||
+				StringUtils.isEmpty(password)) {
+			LOG.error("Missing Constructor arguments. Could not create " +
+				"the GeoServerRESTImporter.");
+		}
+
+		ObjectMapper mapper = new ObjectMapper();
+		mapper.configure(DeserializationFeature.UNWRAP_ROOT_VALUE, true);
+		mapper.configure(SerializationFeature.WRAP_ROOT_VALUE, true);
+		mapper.setSerializationInclusion(Include.NON_NULL);
+
+		this.username = username;
+		this.password = password;
+
+		this.mapper = mapper;
+		this.baseUri = new URI(importerBaseURL);
+	}
+
+	/**
+	 *
+	 * @param importJob
```

Comment:
wrong javadoc parameters

Reply:
Fixed in b119a96e9828b8def01eddd0fd0dd3ede1fe207a.